langchain4j-core/src/main/java/dev/langchain4j/internal/Json.java

```diff
@@ -90,4 +100,15 @@ public static <T> T fromJson(String json, Class<T> type) {
     public static InputStream toInputStream(Object o, Class<?> type) throws IOException {
         return CODEC.toInputStream(o, type);
     }
+
+    /**
+     * Convert the given JSON string to an object of the given type.
+     * @param json the JSON string.
+     * @param type the type of the object represented by {@link TypeToken}.
+     * @return the object.
+     * @param <T> the type of the object.
+     */
+    public static <T> T fromJson(String json, TypeToken<T> type) {
```

Comment:
`TypeToken` is a Gson-specific class, but this interface is supposed to be implementation-agnostic. We are also slowly migrating to Jackson, so ideally the solution should work for Jackson as well.
Is it possible to make it work without introducing `TypeToken` into this interface?

Reply:
![image](https://github.com/user-attachments/assets/742736f0-c370-4958-907a-c0dba5081ae0)
`argument` here is a `List<LinkedTreeMap>`. I guess each `LinkedTreeMap` can (separately) be converted into a `Person` using existing  `Json.fromJson(String json, Class<T> type)` method

Reply:
Tried it but no luck. For Jackson we need to use TypeReference

Reply:
@gs-snagaraj `TypeReference` is needed only when we try to deserialize `List<T>`, right? To deserialize each `T` separately we do not need it.
AFAIU, since `argument` is a `List<T>` where each `T` is `LinkedTreeMap`, we can serialize `LinkedTreeMap` into `String` and deserialize it into T without the need of `TypeReference` or `TypeToken`.